pkg/component/component.go

```diff
@@ -564,51 +572,80 @@ func ApplyConfig(client *occlient.Client, componentConfig config.LocalConfigInfo
 }
 
 // ApplyConfigDeleteURL applies url config deletion onto component
-func applyConfigDeleteURL(client *occlient.Client, componentConfig config.LocalConfigInfo) (err error) {
-
-	urlList, err := urlpkg.ListPushed(client, componentConfig.GetName(), componentConfig.GetApplication())
-	if err != nil {
-		return err
-	}
-	localURLList := componentConfig.GetURL()
-	for _, u := range urlList.Items {
-		if !checkIfURLPresentInConfig(localURLList, u.Name) {
-			err = urlpkg.Delete(client, u.Name, componentConfig.GetApplication())
-			if err != nil {
-				return err
+func applyConfigDeleteURL(client *occlient.Client, kClient *kclient.Client, componentConfig config.LocalConfigInfo, envSpecificInfo envinfo.EnvSpecificInfo) (err error) {
+	if experimental.IsExperimentalModeEnabled() {
+		componentName := envSpecificInfo.GetName()
+		ingressList, err := urlpkg.ListPushedIngress(kClient, componentName)
```

Comment:
I don't think we need it. We can do something like what you did in ApplyConfigCreateURL(). Just pass the `pushedURLMap` here and check against it. Or am I missing something?